common/src/main/java/com/wynntils/models/players/GuildModel.java

```diff
@@ -4,54 +4,108 @@
  */
 package com.wynntils.models.players;
 
+import com.wynntils.core.WynntilsMod;
 import com.wynntils.core.components.Model;
 import com.wynntils.core.text.StyledText;
+import com.wynntils.handlers.chat.event.ChatMessageReceivedEvent;
 import com.wynntils.models.character.CharacterModel;
 import com.wynntils.models.players.type.GuildRank;
 import com.wynntils.utils.mc.LoreUtils;
+import com.wynntils.utils.mc.McUtils;
 import java.util.List;
+import java.util.Locale;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
 
 public class GuildModel extends Model {
-    private static final Pattern GUILD_NAME_MATCHER = Pattern.compile("§3(.*?)§b.*");
-    private static final Pattern GUILD_RANK_MATCHER = Pattern.compile("§7Rank: §f(.*)");
+    // Test suite: https://regexr.com/7hob9
+    private static final Pattern GUILD_NAME_MATCHER = Pattern.compile("§3([a-zA-Z ]*?)§b \\[[a-zA-Z]{3,4}]");
 
-    private String name = "";
-    private GuildRank rank;
+    // Test suite: https://regexr.com/7hobf
+    // Recruiter is intentionally first, otherwise Recruit will incorrectly match first
+    // This applies to all other patterns in this class involving guild ranks
+    private static final Pattern GUILD_RANK_MATCHER =
+            Pattern.compile("§7Rank: §f(Recruiter|Recruit|Captain|Strategist|Chief|Owner)");
```

Comment:
I think if you use the start/end of line markers like this, it will work in the normal rank order:

```
            Pattern.compile("^§7Rank: §f(Recruit|Recruiter|Captain|Strategist|Chief|Owner)$");
```

Reply:
Yeah it does, but this does not work for the MSG_RANK_CHANGED regex. Do you want to use ^$ or just keep it consistent and do recruiter first?

Reply:
Why does it not? It looks like it should work just as well there.

Reply:
If you have no end of line marker, then the last Recruit will match before Recruiter, but if you end with `$` it should be all fine.

Reply:
(As a side not: @kristofbolyai This is one of the reasons why I prefer to always use the explicit `^...$` construct)

Reply:
Oh yeah the first ncg doesn't matter. Okay I will change it.